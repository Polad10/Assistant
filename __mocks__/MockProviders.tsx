import { Dispatch, PropsWithChildren, SetStateAction } from 'react'
import { Translator } from '../helpers/Translator'
import { AppointmentRequest } from '../modals/Appointment'
import { PatientRequest } from '../modals/Patient'
import { Setting } from '../modals/Setting'
import { TreatmentRequest } from '../modals/Treatment'
import { PaymentRequest } from '../modals/Payment'
import { Api } from '../helpers/Api'
import { mockModels } from './MockModels'
import { AuthContext } from '../contexts/AuthContext'
import { DataContext } from '../contexts/DataContext'
import { ThemeContext } from '../contexts/ThemeContext'
import { LocalizationContext } from '../contexts/LocalizationContext'

const themeContextMock = {
  primary: '#121212',
  secondary: '#1a1a1a',
  accent: '#7600eb',
  neutral: '#faf8ff',
  border: '#374151',
  info: '#d3fbd8',
  success: '#00e5b0',
  warning: '#9f1239',
  error: '#9f1239',
}

const localizationContextMock = {
  translator: new Translator('en'),
  language: 'en',
}

const authContextMock = {
  user: undefined,
  loading: false,
}

const dataContextMock = {
  fetchPatients: async () => {},
  fetchAppointments: async () => {},
  fetchTreatments: async () => {},
  fetchPayments: async () => {},
  fetchSetting: async () => {},

  createAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  updateAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  deleteAppointment: async (appointmentId: number) => {},

  createPatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  updatePatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  deletePatient: async (patientId: number) => {},

  createTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  updateTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  deleteTreatment: async (treatmentId: number) => {},

  createPayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  updatePayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  deletePayment: async (paymentId: number) => {},

  createSetting: async (setting: Setting) => {},
  updateSetting: async (setting: Setting) => {},

  api: undefined,
  setApi: (action: SetStateAction<Api | undefined>): Dispatch<SetStateAction<Api | undefined>> => jest.fn(),

  loading: false,
  setLoading: (action: SetStateAction<boolean>): Dispatch<SetStateAction<boolean>> => jest.fn(),

  patients: [mockModels.patient_1, mockModels.patient_2],
  appointments: [mockModels.appointment_1, mockModels.appointment_2],
  treatments: [mockModels.treatment_1, mockModels.treatment_2],
  payments: [mockModels.payment_1, mockModels.payment_2],
  setting: mockModels.setting,
}

export default function MockProviders({ children }: PropsWithChildren) {
  return (
    <AuthContext.Provider value={authContextMock}>
      <DataContext.Provider value={dataContextMock}>
        <ThemeContext.Provider value={themeContextMock}>
          <LocalizationContext.Provider value={localizationContextMock}>{children}</LocalizationContext.Provider>
        </ThemeContext.Provider>
      </DataContext.Provider>
    </AuthContext.Provider>
  )
}
