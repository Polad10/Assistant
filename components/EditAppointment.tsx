import { Mode } from '../enums/Mode';
import Appointment from './Appointment';

export default function EditAppointment({ route }: any) {
  const { treatment } = route.params;

  return <Appointment mode={Mode.EDIT} treatment={treatment} />;
}
