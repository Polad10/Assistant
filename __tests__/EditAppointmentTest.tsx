import renderer from 'react-test-renderer'
import MockProviders from '../__mocks__/MockProviders'
import EditAppointment from '../components/EditAppointment'

describe('EditAppointment', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MockProviders>
          <EditAppointment />
        </MockProviders>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
