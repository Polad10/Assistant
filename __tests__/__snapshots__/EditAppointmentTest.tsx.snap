// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditAppointment renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#121212",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "backgroundColor": "#121212",
          "flex": 1,
        },
        {
          "paddingHorizontal": 10,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flex": 1,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          {
            "flexGrow": 1,
          }
        }
        onScroll={[Function]}
        scrollEventThrottle={50}
      >
        <View>
          <View
            style={
              {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                  "marginRight": 5,
                }
              }
            >
              <View
                style={
                  {
                    "paddingHorizontal": 0,
                    "width": "100%",
                  }
                }
                testID="RNE__Input__view-wrapper"
              >
                <Text
                  style={
                    {
                      "color": "#d3fbd8",
                      "fontSize": 16,
                      "fontWeight": "bold",
                      "marginBottom": 5,
                    }
                  }
                >
                  Date
                </Text>
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "borderColor": "#374151",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "marginBottom": 20,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <TextInput
                    autoCorrect={false}
                    editable={true}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="2024-01-01"
                    placeholderTextColor="#86939e"
                    returnKeyType="done"
                    selectionColor="#faf8ff"
                    showError={false}
                    style={
                      {
                        "color": "#faf8ff",
                        "flex": 1,
                        "fontSize": 18,
                        "minHeight": 40,
                        "paddingHorizontal": 10,
                        "paddingTop": 20,
                        "paddingVertical": 20,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                    value="2024-01-01"
                  />
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "height": 40,
                        "justifyContent": "center",
                        "marginRight": 10,
                        "marginVertical": 4,
                        "paddingRight": 0,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <Text
                  style={
                    {
                      "color": "#ff190c",
                      "fontSize": 12,
                      "height": 0,
                      "margin": 0,
                      "padding": 0,
                    }
                  }
                />
              </View>
              <Modal
                animationType="none"
                hardwareAccelerated={false}
                isVisible={false}
                onHide={[Function]}
                transparent={true}
                visible={false}
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "marginLeft": 5,
                }
              }
            >
              <View
                style={
                  {
                    "paddingHorizontal": 0,
                    "width": "100%",
                  }
                }
                testID="RNE__Input__view-wrapper"
              >
                <Text
                  style={
                    {
                      "color": "#d3fbd8",
                      "fontSize": 16,
                      "fontWeight": "bold",
                      "marginBottom": 5,
                    }
                  }
                >
                  Time
                </Text>
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "borderColor": "#374151",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "marginBottom": 20,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <TextInput
                    autoCorrect={false}
                    editable={true}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    placeholder="23:00"
                    placeholderTextColor="#86939e"
                    returnKeyType="done"
                    selectionColor="#faf8ff"
                    showError={false}
                    style={
                      {
                        "color": "#faf8ff",
                        "flex": 1,
                        "fontSize": 18,
                        "minHeight": 40,
                        "paddingHorizontal": 10,
                        "paddingTop": 20,
                        "paddingVertical": 20,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                    value="20:44"
                  />
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "height": 40,
                        "justifyContent": "center",
                        "marginRight": 10,
                        "marginVertical": 4,
                        "paddingRight": 0,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
                <Text
                  style={
                    {
                      "color": "#ff190c",
                      "fontSize": 12,
                      "height": 0,
                      "margin": 0,
                      "padding": 0,
                    }
                  }
                />
              </View>
              <Modal
                animationType="none"
                hardwareAccelerated={false}
                isVisible={false}
                onHide={[Function]}
                transparent={true}
                visible={false}
              />
            </View>
          </View>
          <View
            style={
              {
                "paddingHorizontal": 0,
                "width": "100%",
              }
            }
            testID="RNE__Input__view-wrapper"
          >
            <Text
              style={
                {
                  "color": "#d3fbd8",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "marginBottom": 5,
                }
              }
            >
              Actions
            </Text>
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "borderBottomWidth": 1,
                  "borderColor": "#374151",
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "marginBottom": 20,
                  "transform": [
                    {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <TextInput
                autoCorrect={false}
                editable={true}
                multiline={true}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Teeth cleaning, dental crowns placement."
                placeholderTextColor="#86939e"
                returnKeyType="default"
                selectionColor="#faf8ff"
                showError={false}
                style={
                  {
                    "color": "#faf8ff",
                    "flex": 1,
                    "fontSize": 18,
                    "minHeight": 100,
                    "paddingHorizontal": 10,
                    "paddingTop": 15,
                    "paddingVertical": 20,
                  }
                }
                testID="RNE__Input__text-input"
                underlineColorAndroid="transparent"
              />
            </View>
            <Text
              style={
                {
                  "color": "#ff190c",
                  "fontSize": 12,
                  "height": 0,
                  "margin": 0,
                  "padding": 0,
                }
              }
            />
          </View>
          <RNGestureHandlerButton
            collapsable={false}
            delayLongPress={600}
            enabled={true}
            exclusive={true}
            handlerTag={1}
            handlerType="NativeViewGestureHandler"
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            touchSoundDisabled={false}
          >
            <View
              accessible={true}
              collapsable={false}
              style={{}}
            >
              <View
                style={
                  {
                    "paddingHorizontal": 0,
                    "width": "100%",
                  }
                }
                testID="RNE__Input__view-wrapper"
              >
                <Text
                  style={
                    {
                      "color": "#d3fbd8",
                      "fontSize": 16,
                      "fontWeight": "bold",
                      "marginBottom": 5,
                    }
                  }
                >
                  Treatment
                </Text>
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "borderBottomWidth": 1,
                      "borderColor": "#374151",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "flexDirection": "row",
                      "marginBottom": 20,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <TextInput
                    autoCorrect={false}
                    editable={false}
                    onBlur={[Function]}
                    onFocus={[Function]}
                    placeholder="Select a treatment"
                    placeholderTextColor="#86939e"
                    pointerEvents="none"
                    returnKeyType="done"
                    selectionColor="#faf8ff"
                    showError={false}
                    style={
                      {
                        "color": "#faf8ff",
                        "flex": 1,
                        "fontSize": 18,
                        "minHeight": 40,
                        "opacity": 0.5,
                        "paddingHorizontal": 10,
                        "paddingTop": 20,
                        "paddingVertical": 20,
                      }
                    }
                    testID="RNE__Input__text-input"
                    underlineColorAndroid="transparent"
                    value="Title"
                  />
                </View>
                <Text
                  style={
                    {
                      "color": "#ff190c",
                      "fontSize": 12,
                      "height": 0,
                      "margin": 0,
                      "padding": 0,
                    }
                  }
                />
              </View>
            </View>
          </RNGestureHandlerButton>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <RCTSafeAreaView>
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 2,
          },
          undefined,
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": false,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginHorizontal": 10,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 2,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingHorizontal": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#9f1239",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Delete
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
